core/domain/skill_services.py

```diff
@@ -514,3 +514,86 @@ def save_skill_summary(skill_summary):
     )
 
     skill_summary_model.put()
+
+
+def create_skill_mastery(user_id, skill_id, degree_of_mastery):
+    """Creates and stores skill mastery of a user.
+
+    Args:
+        user_id: str. The user id of the logged in user.
+        skill_id: str. The unique id of the skill.
+        degree_of_mastery: float. The degree of mastery of user in the skill.
+    """
+    skill_mastery_model = skill_models.SkillsMasteryModel(
+        id=skill_domain.SkillsMastery.get_model_id(user_id, skill_id),
+        user_id=user_id,
+        skill_id=skill_id,
+        degree_of_mastery=degree_of_mastery
+    )
+
+    skill_mastery_model.put()
+
+
+def get_skill_mastery(user_id, skill_id):
+    """Fetches the mastery of user in a particular skill.
+
+    Args:
+        user_id: str. User Id of the user currently logged in.
+        skill_id: str. Unique id of the skill for which mastery degree is
+            requested.
+
+    Returns:
+        degree_of_mastery: float. Mastery degree of the user for the requested
+            skill.
+    """
+    degree_of_mastery = skill_models.SkillsMasteryModel.get(
+        skill_domain.SkillsMastery.get_model_id(
+            user_id, skill_id)).degree_of_mastery
+
+    return degree_of_mastery
+
+
+def get_multi_skill_mastery(user_id, skill_ids):
+    """Fetches the mastery of user in multiple skills.
+
+    Args:
+        user_id: str. User Id of the user currently logged in.
+        skill_ids: list(str). Skill Ids of the skill for which mastery degree is
+            requested.
+
+    Returns:
+        degree_of_mastery: list(float). Mastery degree of the user for requested
+        skills.
+    """
+    degree_of_mastery = []
+    skills_summary = []
+    model_ids = []
+
+    for skill_id in skill_ids:
+        model_ids.append(
+            skill_domain.SkillsMastery.get_model_id(user_id, skill_id))
+
+    skills_summary = skill_models.SkillsMasteryModel.get_multi(model_ids)
+    for skill_summary in skills_summary:
+        degree_of_mastery.append(skill_summary.degree_of_mastery)
+
+    return degree_of_mastery
+
+
+def get_all_skill_mastery(user_id):
+    """Fetches the mastery of user for all skills.
+
+    Args:
+        user_id: str. User Id of the user currently logged in.
+
+    Returns:
+        skill_mastery: dict. Mastery degree of the user for every skill.
```

Comment:
Describe internal structure of dict. What are types and descriptions of the keys/values?